### PR TITLE
refactored evidence levels for teachers and students

### PR DESCRIPTION
**Pull request by kobryan0619**
This refactors the `EvidenceLevels` component by creating two different components - one for teacher view and another for  student view.  For the student view, the students will see a horizontal view of the evidence levels. 

The new student view also works for when text is really long and is responsive to screen resizing:
image

Here's what the teacher side looks like for the various states:
image

## Links

[AITT-219](https://codedotorg.atlassian.net/jira/software/projects/AITT/boards/70?selectedIssue=AITT-219)

## Testing story

Modified one testing file and added two testing files.

### CODE REVIEW

**Comment by kobryan0619** [comment]
I know I have to do a bit of tweaking of the CSS here to get the line to the right of the higlighted box to show up more clearly... that "tweak" is taking more time than it is kinda worth at this point.  I wanted to get this PR out there so you all can review the rest of the work and then either merge this in with a fix after launch OR, If I can figure it out in the next few hours, add it to this PR then.

File: apps/src/templates/rubrics/EvidenceLevels.jsx [hunk 1]
```diff
@@ -25,61 +17,20 @@ export default function EvidenceLevels({
     return newArray.sort((a, b) => b.understanding - a.understanding);
   };
   if (canProvideFeedback) {
```
Comment (bethanyaconnor) [inline apps/src/templates/rubrics/EvidenceLevels.jsx]: `canProvideFeedback` is false on the non-project level or when not viewing a student. We might have to pass in whether the user is a student or teacher here. It helps that students should be seeing `StudentRubricView` while teachers should be seeing `RubricContent`

**Review by bethanyaconnor — COMMENTED** [review]
So speedy! This is a great start but I'm concerned this will have side effects in other views. I unfortunately broke the `RubricContainer` storybook (got a fix coming soon) but it might be worth looking at the `LearningGoal` storybook and making sure the views are expected.

File: apps/src/templates/rubrics/EvidenceLevels.jsx (hunk 1 above)
Comment (kobryan0619) [reply apps/src/templates/rubrics/EvidenceLevels.jsx]: I need the face-palm emoji here... I knew that!  I ended up using `isStudent` from the student view which was easier than trying to navigate passing through a prop that uses the teacher indicator.

File: apps/src/templates/rubrics/EvidenceLevelsForTeachers.jsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+} from '@cdo/apps/componentLibrary/typography';
+import {UNDERSTANDING_LEVEL_STRINGS} from './rubricHelpers';
+
+export default function EvidenceLevelsForTeachers({
+  evidenceLevels,
+  learningGoalKey,
+  understanding,
+  radioButtonCallback,
+  canProvideFeedback,
+  submittedEvaluation,
```
Comment (bethanyaconnor) [inline apps/src/templates/rubrics/EvidenceLevelsForTeachers.jsx]: can `submittedEvaluation` ever be non-null for a teacher?

File: apps/src/templates/rubrics/LearningGoal.jsx [hunk 3]
```diff
@@ -35,6 +35,7 @@ export default function LearningGoal({
   aiUnderstanding,
   aiConfidence,
   submittedEvaluation,
+  isStudent,
```
Comment (bethanyaconnor) [inline apps/src/templates/rubrics/LearningGoal.jsx:38]: nit: can you explicitly set this to false in `RubricContent`?

File: apps/test/unit/templates/rubrics/EvidenceLevelsTest.jsx [hunk 4]
```diff
@@ -14,62 +12,28 @@ const DEFAULT_PROPS = {
...
-      lastEvidenceLevel.teacherDescription
-    );
-    expect(wrapper.find('Memo(RadioButton)').at(0).prop('label')).to.equal(
-      UNDERSTANDING_LEVEL_STRINGS[lastEvidenceLevel.understanding]
-    );
+  it('renders teachers view of evidence levels when the user can provide feedback', () => {
+    const wrapper = shallow(<EvidenceLevels {...DEFAULT_PROPS} />);
+    expect(wrapper.find('EvidenceLevelsForTeachers').length).to.equal(1);
+    expect(
+      wrapper.find('EvidenceLevelsForTeachers').props().canProvideFeedback
```
Comment (bethanyaconnor) [inline apps/test/unit/templates/rubrics/EvidenceLevelsTest.jsx:19]: why is this undefined? The name of this test suggests it should be true?

File: apps/test/unit/templates/rubrics/EvidenceLevelsForTeachersTest.jsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+const DEFAULT_PROPS = {
+  evidenceLevels: [
+    {id: 1, understanding: 0, teacherDescription: 'test1'},
+    {id: 2, understanding: 1, teacherDescription: 'test2'},
+  ],
+  learningGoalKey: 'key-1',
+};
+
+describe('EvidenceLevels', () => {
```
Comment (bethanyaconnor) [inline apps/test/unit/templates/rubrics/EvidenceLevelsForTeachersTest.jsx]: nit: should be `EvidenceLevelsForTeachers`

File: apps/test/unit/templates/rubrics/EvidenceLevelsTest.jsx (hunk 4 above)
Comment (kobryan0619) [reply apps/test/unit/templates/rubrics/EvidenceLevelsTest.jsx:19]: That test should have been renamed to "renders teachers view of evidence levels when the user can **not** provide feedback".  It felt weird to test that something was `undefined` but I checked to makes sure that we do it in other places in our codebase (which we do).

File: apps/src/templates/rubrics/EvidenceLevelsForTeachers.jsx (hunk 2 above)
Comment (kobryan0619) [reply apps/src/templates/rubrics/EvidenceLevelsForTeachers.jsx]: You're correct about this.  I removed it here.  This was a hold over from when this was all in one file that I missed.

File: apps/test/unit/templates/rubrics/EvidenceLevelsTest.jsx (hunk 4 above)
Comment (bethanyaconnor) [reply apps/test/unit/templates/rubrics/EvidenceLevelsTest.jsx:19]: it does feel a little weird but it makes _a lot_ more sense with the updated test name. Thanks for updating!

**Review by bethanyaconnor — APPROVED** [review]
🎉

**Pull request merged**